src/tin.c

```diff
@@ -108,6 +109,7 @@ int main(int argc, char **argv)
 		print_step("Parsing and preprocessing failed\n");
 		goto fail;
 	}
+	optimize(mod, mod->ast_root);
```

Comment:
The code currently cannot compile some example files, such as examples/include.tin, and instead segfaults. Omitting the call to optimize results in it compiling successfully, so there must be an issue within the optimization code.